queues/c-or-cpp/double-ended-queue-using-array.c

```diff
@@ -0,0 +1,192 @@
+    printf("queue is full");
+    //checking is the rear equal to -1 means queue is empty or not
+else if(array->rear==-1)
+	{
+	    array->rear++;
+	    array->arr[array->rear]=data;
+	    array->front=array->rear;
+	}
+	// checking whether queue is full from rear but not from front
+	else if(array->rear==array->capacity-1&&array->front!=0)
+	{
+	    array->rear=0;
+	    array->arr[array->rear]=data;
+	}
+	// normal condition
+	else
+	{
+	    array->rear++;
+	    array->arr[array->rear]=data;
+	}
+
+}
+// deletion front basically delete data from front of queue
+void deletion_front(struct deque *array)
+{
+    // queue empty condition
+    if(array->front==-1)
+	printf("queue is empty");
+    else
+    {
+        // printing the value which is going to be deleted
+    printf("deleted value is %d",array->arr[array->front]);
+    // checking whether the queue has only one data
+	if(array->front==array->rear)
+	{
+	    array->front=-1;
+	    array->rear=-1;
+	}
+	// checking whether front is equal to capacity-1
+	else if(array->front==array->capacity-1)
+	    array->front=0;
+	// normal condition
+	else
+	    array->front++;
+
+
+ }
+}
+// deletion rear basically delete data from end of queue
+void deletion_rear(struct deque *array)
+{
+    //empty condition
+    if(array->rear==-1)
+      printf("queue is empty");
+      else
+      { printf("deleted value is %d",array->arr[array->rear]);
+    // checking whether the queue has only one data
+          if(array->rear==array->front)
+          {
+        array->front=-1;
+	    array->rear=-1;
+          }
+          // checking whether rear is 0 but there not only one data left that means that data are at end of queue
+          else if(array->rear==0)
+            array->rear=array->capacity-1;
+            //normal condition
+          else
+            array->rear--;
+      }
+}
+// menu basically provide the no. of task can be performed so you can choose one of them
+int menu(void)
+{
+    int choice;
+    printf("\n1: Insertion at Front");
+    printf("\n2: Insertion at Rear");
+    printf("\n3: Deletion at Front");
+    printf("\n4: Deletion at Rear");
+    printf("\n5: exit");
+    printf("\n\n enter your choice");
+    scanf("%d",&choice);
+    return(choice);
+}
+// driver code
+int main()
+{
+    struct deque *array;
+    int cap,data;
+    printf("enter the capacity");
+    scanf("%d",&cap);
+    array=create_dequeue(cap);
+    while(1)
+    { system("cls");
+        switch(menu())
+        {
+        case 1:
+            printf("enter value");
+            scanf("%d",&data);
+            insertion_front(array,data);
+            break;
+        case 2:
+            printf("enter value ");
+            scanf("%d",&data);
+            insertion_rear(array,data);
+            break;
+        case 3:
+            deletion_front(array);
+            break;
+        case 4:
+            deletion_rear(array);
+            break;
+        case 5:
```

Comment:
### Optional
it would be amazing if you could add a function to see the current status of the queue. ( something like `display_queue()` function )

Reply:
ok i will do all the recommended changes

Reply:
can i create display_queue(); function later on..

Reply:
@UG-SEP not a problem, you are always welcome to add more functionality to already submitted programs later down the line.